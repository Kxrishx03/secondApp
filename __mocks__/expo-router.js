//mock push,navigate,replace,useLocalSearchParams

const mockPush = jest.fn();
const mockReplace = jest.fn();
const mockNavigate = jest.fn();
const mockUseLocalSearchParams = jest.fn();

const mockRouter = {
    push:mockPush,
    replace:mockReplace,
    naviagte:mockNavigate,

}

export const useRouter = () => (mockRouter);
export const push = mockPush;
export const replace = mockReplace;
export const naviagte = mockNavigate;
export const  useLocalSearchParams = mockUseLocalSearchParams;