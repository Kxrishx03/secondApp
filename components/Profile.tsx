import { View, Text } from 'react-native';
import React from 'react';
import { useLocalSearchParams } from 'expo-router';

export default function () {
    const { id } = useLocalSearchParams<{ id?: string }>()
    return (
        <View>
            <Text testID='profile-id'>{id ? `Profile ID: ${id}` : 'No Profile ID provided'}</Text>
        </View>

    );
  
}