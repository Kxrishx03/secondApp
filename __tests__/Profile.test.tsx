import React from 'react'; 
import { render ,screen, fireEvent, renderHook, act} from '@testing-library/react-native';
import Profile from '@/components/Profile';
import { useLocalSearchParams } from 'expo-router';
import renderer from 'react-test-renderer';


jest.mock('expo-router');
describe('Testing UseLocalSearchParams',()=>{
       
    test('Test when id is provided',()=>{
          //MOCK 
          (useLocalSearchParams as jest.Mock).mockReturnValue({id:'123'});
          //RENDER
         const {getByText} = render(<Profile />);
          //CHECKS
          expect(getByText('Profile ID: 123')).toBeTruthy();
    });

    test('Test when id is not  provided',()=>{
        //MOCK 
        (useLocalSearchParams as jest.Mock).mockReturnValue({id:null});
        //RENDER
       const {getByText} = render(<Profile />);
        //CHECKS
        expect(getByText('No Profile ID provided')).toBeTruthy();
  });

   //Snapshot Testing
      test('Snapshot practice',()=>{
        const tree = renderer.create(<Profile />).toJSON();
        expect(tree).toMatchSnapshot();
      })
})