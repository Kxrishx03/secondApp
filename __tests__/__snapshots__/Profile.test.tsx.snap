// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UseLocalSearchParams Snapshot practice 1`] = `
<View>
  <Text
    testID="profile-id"
  >
    No Profile ID provided
  </Text>
</View>
`;
